lnrpc/marshall_utils.go

```diff
@@ -179,3 +180,57 @@ func MarshalOutPoint(op *wire.OutPoint) *OutPoint {
 		OutputIndex: op.Index,
 	}
 }
+
+// Parse the coin selection strategy string from the command line
+// to an equivalent lnrpc.CoinSelectionStrategy enum type.
+func ParseCoinSelectionStrategyString(strategy string)(
+	*CoinSelectionStrategy, error) {
+
+	var coinSelectionStrategy CoinSelectionStrategy
+	switch strategy {
+	case "global-config":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG
+
+	case "largest":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_LARGEST
+
+	case "random":
+		coinSelectionStrategy = CoinSelectionStrategy_STRATEGY_RANDOM
+
+	default:
+		return nil, fmt.Errorf("unknown coin selection strategy "+
+			"%v", strategy)
+	}
+
+	return &coinSelectionStrategy, nil
+}
+
+// Parse the coins selection strategy lnrpc.CoinSelectionStrategy enum type
+// to an equivalent wallet.CoinSelectionStrategy type.
+func ParseCoinSelectionStrategyEnum(strategy CoinSelectionStrategy)(
+	wallet.CoinSelectionStrategy, error) {
+
+	var coinSelectionStrategy wallet.CoinSelectionStrategy
+
+	switch strategy {
+	case CoinSelectionStrategy_STRATEGY_USE_GLOBAL_CONFIG:
+		strategyValue, _ := ParseCoinSelectionStrategyString(
+			"largest",
```

Comment:
@guggero Thanks a lot for your feedback the code looks much cleaner because of the usage of common parsing functions. Now something there is one thing left here in `lnrpc/marshall_utils.go` when I use `lnd.DefaultConfig().CoinSelectionStrategy` instead of `largest` it gives me Import Cycle dependency issue because I needed to import `github.com/lightningnetwork/lnd` at the top of the file. I'm not used much to these issues any suggestions would be helpful, Thanks 🙏

Reply:
That's why I suggested to pass in the configured strategy as a parameter. Then the caller will find out what it is (`lnd.DefaultConfig().CoinSelectionStrategy` is wrong anyway, because that's not what the user configured but the default value if nothing is set by the user).